### DISCUSSION TITLE
Is there a way to cancel the "persist" with programmatically?

### DISCUSSION THREAD

**Discussion by bozdevs** [swup/swup#975] [Help]
In some cases I need all `data-swup-persist` to be ignored, for example "language switch".

For this, I wrote some coding using hooks and tried to show them as separate containers. You can find the code I made below:

```js
          try {
            var visit = event.detail.visit,
              element = visit.trigger.el;

            if (element && element.hasAttribute("data-force-persists")) { // If link contains "data-force-persists"

              var forces = document.querySelectorAll("[data-swup-persist]");

              forces.forEach(function (force) {
                var value = force.getAttribute("data-swup-persist");
                if (value) {
                  visit.containers.push("[data-swup-persist='" + value + "']");
                }
              });

            }
          } catch (error) {
            console.warn(error);
          }
```

This way it adds the container properly as `[data-swup-persist="unique-key"]`, but it cannot "Override" it.

Is there another way to do this?

----

I have experienced that if I remove the `[data-swup-persist]` attribute it works. But I am not sure how accurate this approach is. Because theoretically if the response is before these removals it will not work. Do you think this approach is accurate enough?

```js
          try {
            var visit = event.detail.visit,
              element = visit.trigger.el;

            if (element && element.hasAttribute("data-force-persists")) {

              var forces = document.querySelectorAll("[data-swup-persist]");

              forces.forEach(function (force) {
                force.removeAttribute("data-swup-persist");
                // var value = force.getAttribute("data-swup-persist");
                // if (value) {
                //   visit.containers.push("[data-swup-persist='" + value + "']");
                // }
              });

            }
          } catch (error) {
            console.warn(error);
          }
```

**Comment by daun** [comment]
As I wrote in a previous thread, the persist attribute is meant for single elements to maintain their internal state in JS. It's not an alternative to using containers. So using it the way you're doing, I'd suspect it to break in subtle ways that would be hard to debug. Why not switch to using containers only? You can still set them dynamically depending on whatever conditions you're checking against.

**Comment by daun** [comment]
To answer your question more directly, though, there's currently no feature in swup to globally disable persisting elements. But you could probably remember which elements had the attribute before the content update (maybe in `visit:start`), remove the attributes, and restore them after the content update (in `page:view`).
